alicloud/data_source_alicloud_cen_instances.go

```diff
@@ -0,0 +1,236 @@
+		args.PageNumber = requests.NewInteger(pageNumber)
+	}
+
+	if len(allCens) < 1 {
+		return fmt.Errorf("Your query returned no results. Please change your search criteria and try again.")
+	}
+
+	log.Printf("[DEBUG] alicloud_cen_instances - Cens found: %#v", allCens)
+	return censDescriptionAttributes(d, allCens, meta)
+}
+
+func constructCenRequestFilters(d *schema.ResourceData) ([]cbn.DescribeCensFilter, error) {
+
+	var filters []cbn.DescribeCensFilter
+	for _, key := range []string{"cen_ids", "cen_bandwidth_package_ids"} {
+		if v, ok := d.GetOk(key); ok && len(v.([]interface{})) > 0 {
+			var filterElem []string
+			for _, vv := range v.([]interface{}) {
+				if vv == nil {
+					continue
+				}
+				filterElem = append(filterElem, Trim(vv.(string)))
+			}
+			if len(filterElem) > 0 {
+				filters = append(filters, cbn.DescribeCensFilter{
+					Key:   terraformToAPI(key[:len(key)-1]),
+					Value: &filterElem,
+				})
+			}
+		}
+	}
+
+	return filters, nil
+}
+
+func censDescriptionAttributes(d *schema.ResourceData, cenSetTypes []cbn.Cen, meta interface{}) error {
+	var ids []string
+	var s []map[string]interface{}
+
+	for _, cen := range cenSetTypes {
+		mapping := map[string]interface{}{
+			"id":                        cen.CenId,
+			"name":                      cen.Name,
+			"status":                    cen.Status,
+			"cen_bandwidth_package_ids": cen.CenBandwidthPackageIds.CenBandwidthPackageId,
+			"description":               cen.Description,
+		}
+
+		// get child instances
+		instanceIds, err := censDescribeCenAttachedChildInstances(d, cen.CenId, meta)
+		if err != nil {
+			return err
+		}
+		if instanceIds != nil && len(instanceIds) > 0 {
+			mapping["instance_ids"] = instanceIds
+		}
+
+		ids = append(ids, cen.CenId)
+		s = append(s, mapping)
+	}
+
+	d.SetId(dataResourceIdHash(ids))
+	if err := d.Set("cens", s); err != nil {
+		return err
+	}
+
+	// create a json file in current directory and write data source to it.
+	if output, ok := d.GetOk("output_file"); ok && output.(string) != "" {
+		writeToFile(output.(string), s)
+	}
+
+	return nil
+}
+
+func censDescribeCenAttachedChildInstances(d *schema.ResourceData, cenId string, meta interface{}) ([]string, error) {
+
+	var instanceIds []string
+
+	childInstancesReq := cbn.CreateDescribeCenAttachedChildInstancesRequest()
+	childInstancesReq.CenId = cenId
+	childInstancesReq.PageSize = requests.NewInteger(PageSizeLarge)
+
+	for {
+		resp, err := meta.(*AliyunClient).cenconn.DescribeCenAttachedChildInstances(childInstancesReq)
+		if err != nil {
+			return nil, fmt.Errorf("DescribeCenAttachedChildInstances got an error: %#v.", err)
+		}
+		if resp != nil && len(resp.ChildInstances.ChildInstance) > 0 {
+
+			for _, inst := range resp.ChildInstances.ChildInstance {
+				instanceIds = append(instanceIds, inst.ChildInstanceId)
+			}
+
+		}
+
+		if len(resp.ChildInstances.ChildInstance) < PageSizeLarge {
+			break
+		}
+
+		childInstancesReq.PageNumber += requests.NewInteger(1)
```

Comment:
There is a bug in line 232 and you can refer to https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/data_source_alicloud_vpcs.go#L132